### PR TITLE
remote: deprecated unused experimental remote_retry flags

### PR DESCRIPTION
**Pull request by ishikhman**
Part of #7205

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java [hunk 1]
```diff
@@ -119,50 +119,62 @@
...
   @Option(
-      name = "experimental_remote_retry_max_attempts",
+      name = "remote_retries",
+      oldName = "experimental_remote_retry_max_attempts",
       defaultValue = "5",
       documentationCategory = OptionDocumentationCategory.REMOTE,
       effectTags = {OptionEffectTag.UNKNOWN},
-      help = "The maximum number of attempts to retry a transient error.")
-  public int experimentalRemoteRetryMaxAttempts;
+      help = "The maximum number of attempts to retry a transient error."
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: nit: add space

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java [hunk 2]
```diff
@@ -119,50 +119,62 @@
...
       effectTags = {OptionEffectTag.UNKNOWN},
       help = "Whether to retry transient remote execution/cache errors.")
   public boolean experimentalRemoteRetry;
 
+  @Deprecated
   @Option(
       name = "experimental_remote_retry_start_delay_millis",
       defaultValue = "100",
       documentationCategory = OptionDocumentationCategory.REMOTE,
+      deprecationWarning = "Deprecated. See --remote_retries instead.",
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: s/See/Use/

File: src/test/java/com/google/devtools/build/lib/remote/RemoteRetrierTest.java [hunk 3]
```diff
@@ -148,7 +148,7 @@ public void testInterruptedExceptionIsPassedThrough() throws Exception {
     InterruptedException thrown = new InterruptedException();
 
     RemoteOptions options = Options.getDefaults(RemoteOptions.class);
-    options.experimentalRemoteRetry = false;
+    options.remoteRetryMaxAttempts = 0;
```
Comment (buchgr) [inline src/test/java/com/google/devtools/build/lib/remote/RemoteRetrierTest.java]: s/remoteRetryMaxAttempts/remoteMaxRetryAttempts/

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java [hunk 4]
```diff
@@ -119,50 +119,62 @@
...
+  public int remoteRetryMaxAttempts;
 
+  @Deprecated
   @Option(
       name = "experimental_remote_retry_multiplier",
       defaultValue = "2",
       documentationCategory = OptionDocumentationCategory.REMOTE,
+      deprecationWarning = "Deprecated. See --remote_retries instead.",
       effectTags = {OptionEffectTag.UNKNOWN},
       help = "The multiplier by which to increase the retry delay on transient errors.")
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: Update the help message to say that this flag is a no-op and has no effect?

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java [hunk 5]
```diff
@@ -119,50 +119,62 @@
...
+      help = "The maximum number of attempts to retry a transient error."
+          + "If set to 0, retries are disabled.")
+  public int remoteRetryMaxAttempts;
 
+  @Deprecated
   @Option(
       name = "experimental_remote_retry_multiplier",
       defaultValue = "2",
       documentationCategory = OptionDocumentationCategory.REMOTE,
+      deprecationWarning = "Deprecated. See --remote_retries instead.",
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: Update the message to say that this flag has no effect anymore?

**Comment by buchgr** [comment]
LGTM

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java (hunk 2 above)
Comment (ishikhman) [reply src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: done

File: src/test/java/com/google/devtools/build/lib/remote/RemoteRetrierTest.java (hunk 3 above)
Comment (ishikhman) [reply src/test/java/com/google/devtools/build/lib/remote/RemoteRetrierTest.java]: Done

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java (hunk 5 above)
Comment (ishikhman) [reply src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: done for all

**Comment by ishikhman** [comment]
> LGTM. Add a RELNOTES tag maybe?

I will, on the importer CL :)

File: src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java (hunk 4 above)
Comment (ishikhman) [reply src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java]: done

**Pull request closed without merging**